ils_middleware/tasks/sinopia/email.py

```diff
@@ -1,92 +1,89 @@
 """Sinopia Operators and Functions for email notifications."""
 from airflow.providers.amazon.aws.hooks.ses import SESHook
 
-import json
 import logging
 
-from airflow.models.taskinstance import TaskInstance
 from honeybadger import honeybadger
 
 
 logger = logging.getLogger(__name__)
 
 
-# NOTE: Another approach to consider would be to return an EmailOperator, either by
-# using a factory pattern to return a custom operator instance, or by subclassing.
-# see:
-# https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/operators/email/index.html
-# https://airflow.apache.org/docs/apache-airflow/stable/howto/email-config.html?highlight=ses#send-email-using-aws-ses
+def send_notification_emails(**kwargs) -> None:
```

Comment:
@jermnelson I'm wondering if you have an idea on how to test this method? Specifically, i think I just need to mock the `send_upate_success_emails` and `send_task_failure_notifications` methods and assert that they've been called. But this is a bit different than mocking we've done so far. If you have an ideas, please comment here or we can pari.

Reply:
You may have already tried this but could you create a `mocker.Mock` for both `send_upate_success_emails` and `send_task_failure_notifications` within the test function and then assert both have been called with [assert_called_once](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once)?

Reply:
Worked :+1: